File: cpp/codewars-training/trailing_zeros_of_factorial/kata.cpp
#include "trailing_zeros_of_factorial/kata.hpp"

namespace trailing_zeros_of_factorial
{
    auto zeros(const long n) -> long
    {
        long result = 0;
        for (long i = 5; n / i >= 1; i *= 5)
        {
            result += n / i;
        }
        return result;
    }
}

File: dotnet/codewars/Codewars.Training/Codewars.Training.SquaresIntoSquares/Kata.cs
using System.Collections.Generic;

namespace Codewars.Training.SquaresIntoSquares;

public static class Kata
{
    public static string? Decompose(long n)
    {
        var result = DecomposeHelper(n, n * n);
        return result == null ? null : string.Join(" ", result);
    }

    private static List<long>? DecomposeHelper(long n, long remaining)
    {
        if (remaining == 0)
            return new List<long>();

        for (var i = n - 1; i > 0; i--)
        {
            var square = i * i;
            if (remaining - square >= 0)
            {
                var result = DecomposeHelper(i, remaining - square);
                if (result != null)
                {
                    result.Add(i);
                    return result;
                }
            }
        }

        return null;
    }
}